src/WebUI/Common/KubeSummary.tsx

```diff
@@ -94,12 +101,13 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
         this._selectionStore.addChangedListener(this._onSelectionStoreChanged);
 
         this._workloadsActionCreator = ActionsCreatorManager.GetActionCreator<WorkloadsActionsCreator>(WorkloadsActionsCreator);
+        this._selectionActionCreator = ActionsCreatorManager.GetActionCreator<SelectionActionsCreator>(SelectionActionsCreator);
 
         // Ensure workload store is created before get Deployments action
         this._workloadsStore = StoreManager.GetStore<WorkloadsStore>(WorkloadsStore);
 
         // Fetch deployments in parent component we need to show nameSpace in heading and namespace is obtained from deployment metadata
-        this._workloadsActionCreator.getDeployments(this.props.kubeService);
+        this._workloadsActionCreator.getDeployments(KubeSummary._kubeservice);
```

Comment:
we should have a separate class in which we can have a private static, and then we do like getInstance() and then use the same. this way, if someone wants to not to check null/undefined we will be good.

i could getImageService(), why do not you want to use this function?